docs: update and align README

worldofgeese opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first pull request, please read our contributor guidelines in the https://github.com/garden-io/garden/blob/main/CONTRIBUTING.md file.
2. Please label this pull request according to what type of issue you are addressing (see "What type of PR is this?" below)
3. Ensure you have added or run the appropriate tests for your PR.
4. If the PR is unfinished, add `WIP:` at the beginning of the title or use the Github Draft PR feature.
5. Please add at least two reviewers to the PR. Currently active maintainers are: @edvald, @thsig, @eysi09, @Orzelius and @vvagaytsev.
-->

**What this PR does / why we need it**:

Updates the README to include a table of contents, and simplifies and standardizes to [README conventions](https://github.com/dguo/make-a-readme): short description, usage, features, community, contributing, and license. We also include a GIF or two to more directly communicate *the gist*.

The Garden logo has been swapped for one with a background to accommodate users of any GitHub theme (big thanks to Andrea).

Here's the old logo with GitHub's default dark theme:

![Old logo](https://github.com/garden-io/garden/assets/59834693/be38579e-7d6b-40c9-a1de-2dd997fc8a48)

New logo 🍹 :

![New logo](https://github.com/garden-io/garden/assets/59834693/ce51c20b-ab2e-4d01-8cbb-4fd6cee5ba57)

**Notes for reviewer**:

Generally, I've tried to put light on a few juicy features that directly communicate value. We've often heard Garden is just _too_ fantastically powerful and it can be hard to pull out just where to start and what to do. In this update, I've tried to stick to features I know developers will love and can pick up in an afternoon. If you think other features are juicier, please don't hesitate to suggest as replacements.

The original README has been *significantly* slimmed: when reflecting on what to put *back*, I urge a Marie Kondo approach:

![Does this spark joy?](https://github.com/garden-io/garden/assets/59834693/b92223a8-d741-4dec-836e-d956a8e391e6)

thsig commented:
Thoughts on this, @eysi09?

thsig commented:
Looks great! There are a couple of dead links in there (have probably been dead for a while—I think we only run the docs check on changed files).

From the [CI run](https://app.circleci.com/pipelines/github/garden-io/garden/17760/workflows/ec107204-a326-4e86-af4f-4a6d1c1b6c67/jobs/308754):
```
  ERROR: 2 dead links found!
  [✖] https://go.garden.io/garden → Status: 404
  [✖] https://docs.garden.io/basics/installation → Status: 404
```

eysi09 commented:
This is beautiful! Love the simplicity and the terminal screencap. 

Just one thought, should we add a section called Docs that links to them, similar to what we have in the Community section. 

Beside the link at the top, there's not much that references to docs proper (except implicitly in other links) and you might think that's all there is to it.

eysi09 reviewed (changes requested):
This is beautiful! Love the simplicity and the terminal screencap. 

Just one thought, should we add a section called Docs that links to them? Similar to what we have in the Community section.

Beside the link at the top, there's not much that references to docs proper (except implicitly in other links) and you might think that's all there is to it.

shumailxyz commented on README.md:
```diff
@@ -1,5 +1,7 @@
+_If you love Garden, please ★ star this repository to show your support :green_heart:. Looking for support? Join our [Discord](https://go.garden.io/garden)._
```
```suggestion
_If you love Garden, please ★ star this repository to show your support :green_heart:. Looking for support? Join our [Discord](https://go.garden.io/discord)._
```

Orzelius commented on README.md:
```diff
@@ -8,136 +10,122 @@
-With the Stack Graph, each part of your stack describes itself using simple, intuitive YAML declarations, without changing any of your existing code. Garden collects all of your declarations—even across multiple repositories—into a full graph of your stack.
+## Usage
 
-![Stack Graph](docs/stack-graph-drawing.png)
```
I think the stack graph, preferably a picture should still be present. That was the feature that drew me in at first and I think it's the main selling point of Garden in general.

worldofgeese replied:
Pushing back on this a bit: I'm trying for succinct and direct value. The problem with the Stack Graph is it's not something you can do. Can you suggest a way to communicate the value of the Stack Graph that isn't a very large image and doesn't require a lot of explanation?

"Garden's Stack Graph is a directed acyclic graph (uh oh now we have to explain a DAG). To use the Stack Graph (can't do this because it's inherent to Garden's design)..."

You see what I'm getting at? It's a lot of theory. As a user, I want to know a feature I can make use of. If it uses a DAG, that's not that important to me: what matters is the feature works and I can use it.

Orzelius replied:
> The problem with the Stack Graph is it's not something you can do.

Sure, but the graph allows doing things with garden that would otherwise be impossible or way less efficient. Just listing all those things without mentioning the graph sounds like empty promises or wild dreams to me. Literally all cool and useful functionality of garden that separates us from the competition relies on the Graph.

Maybe something like this:
* The stack graph allows you to declare the dependency structure of your project and track changes to avoid unnecessary builds, deploys and test runs. It's like ci/cd config that you can additionally use for development.

eysi09 replied:
I tend to agree with both points. 

How about we change the "Plugins" section to "How Garden Works". Add one paragraph on the Stack Graph, similar to what @Orzelius wrote above and then add the current "Plugins" stuff below? 

In it's current state, the Plugins section starts a bit abruptly with a "how actions are executed..." which might be confusing without the stack graph context.

eysi09 replied:
Something like:

## How Garden Works

Garden users describe their stack with YAML in terms of four actions: Builds, Deploys, Tests, Runs. This allows you to fully codify your workflows, making the reproducible and portable. 

How these actions executed depends on the plugins used. Our Kubernetes plugin is currently the most popular, followed by our Terraform and Pulumi plugins. 

For a deeper dive on the Stack Graph and how Garden works in general, we recommend:

- [How Garden Works](https://docs.garden.io/basics/how-garden-works)
- [A video tour of Garden's directed acyclic graph and getting started](https://www.youtube.com/watch?app=desktop&v=3gMJWGV0WE8)

For a plugin specific docs, checkout:

- [Kubernetes plugin](https://docs.garden.io/guides/remote-kubernetes)
- [Terraform plugin](https://docs.garden.io/terraform-plugin/about)
- [Pulumi plugin](https://docs.garden.io/pulumi-plugin/about)

worldofgeese replied:
@eysi09 I have pushed a commit with those changes. Can you review? 😺

eysi09 replied:
LGTM! Let's shipit.

Orzelius commented on README.md:
```diff
@@ -8,136 +10,122 @@
+## Demo
 
-### **How does it work?**
+![Garden dev deploy](https://raw.githubusercontent.com/ShankyJS/garden-quickstart-content/d8095ad1a8615edf49e721b8afcd901f3056e127/dev-mode.gif)
```
I think this should be a bit faster, maybe speed up the deploy bit. Also running something like `test e2e` after the deploy would show more of garden capabilities.

worldofgeese replied:
This might be a stretch goal: @ShankyJS created it and it's already merged in the quickstart revamp.

Orzelius replied:
With just a deploy it's kinda lame tho, makes it seem like another preview environment provider

worldofgeese replied:
I don't want to make this a merge blocker for release: can we try and improve it in a later PR?

Orzelius replied:
sure

worldofgeese commented on README.md:
```diff
@@ -8,136 +10,122 @@
 
-This is the repo for the open source Garden Core. To learn more about Garden Cloud [click here](https://cloud.docs.garden.io/).
+## Table of Contents
 
-_If you’re using Garden or if you like the project, please ★ star this repository to show your support 💖_
+- [Installation](#installation)
+- [Demo](#demo)
+- [Usage](#usage)
+- [Plugins](#plugins)
+- [Community](#community)
+- [Contributing](#contributing)
+- [License](#license)
```
I googled a built-in ToC feature and couldn't find one. As I understand it, this is the correct way to do a ToC in GitHub-flavored Markdown. This is a ToC for the document itself. The links below the logo link out.

Orzelius replied:
The screenshot I sent is the built in github toc, it's available for any md doc by clicking on the icon on top left

worldofgeese replied:
Sorry: the pic didn't come through on VS Code. I've never once noticed this, which I think is damning of GitHub's implementation. It's totally tucked away visually. I'm not going to die on this hill and am open to clearing out the ToC if you think it detracts from the experience.

worldofgeese replied:
Removed in latest commit

worldofgeese commented:
> Looks great! There are a couple of dead links in there (have probably been dead for a while—I think we only run the docs check on changed files).
> 
> From the [CI run](https://app.circleci.com/pipelines/github/garden-io/garden/17760/workflows/ec107204-a326-4e86-af4f-4a6d1c1b6c67/jobs/308754):
> 
> ```
>   ERROR: 2 dead links found!
>   [✖] https://go.garden.io/garden → Status: 404
>   [✖] https://docs.garden.io/basics/installation → Status: 404
> ```

I've fixed the first dead link but I can't find the other in the README, https://docs.garden.io/basics/installation

shumailxyz commented:
> I've fixed the first dead link but I can't find the other in the README, https://docs.garden.io/basics/installation

@worldofgeese here's the dead link


https://github.com/garden-io/garden/blob/a6ec2899ab57a6afbd4e72bde89a7ea2dbfcf1e7/README.md?plain=1#LL41C2-L41C29

worldofgeese commented:
> > I've fixed the first dead link but I can't find the other in the README, https://docs.garden.io/basics/installation
> 
> @worldofgeese here's the dead link
> 
> https://github.com/garden-io/garden/blob/a6ec2899ab57a6afbd4e72bde89a7ea2dbfcf1e7/README.md?plain=1#LL41C2-L41C29

Fixed, thanks!

eysi09 commented on README.md:
````diff
@@ -8,80 +12,82 @@
+```
 
-Garden then leverages your _existing tools and infrastructure_ to execute this graph, allowing you to go **from zero to a running system in a single command**. And thanks to smart caching, it’s fast!
+Exit with `Ctrl+C` or `exit`.
````
You should exit by typing `exit`, otherwise you see this warning:

```
Syncs started during this session may still be active when this command terminates. You can run garden sync stop '*' to stop all code syncs. Hint: To stop code syncing when exiting garden dev, use Ctrl-D or the exit command.
→ Run garden util hide-warning dev-syncs-active to disable this warning.```

worldofgeese replied:
Fixed, thanks!

eysi09 reviewed (approved):
💯

vvagaytsev reviewed (approved):
🚀
